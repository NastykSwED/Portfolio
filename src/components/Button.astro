---
import { Icon } from 'astro-icon';

interface Props {
	text: string;
	variant?: string;
	type?: 'button' | 'submit' | 'reset' | undefined;
	iconName: string;
	iconPackName: string;
	iconUtility?: string;
}
const { text, type, variant, iconPackName, iconName, iconUtility } =
	Astro.props;
---

<button class={variant ? `Button Button--${variant}` : 'Button'} type={type}
	>{text}
	<span class={`Button-Icon ${iconUtility}`}>
		<Icon pack={iconPackName} name={iconName} class={iconUtility} />
	</span>
</button>
