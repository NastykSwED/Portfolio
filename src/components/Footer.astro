---
const year = new Date().getFullYear();
---

<footer class='re-Footer' id='re-Footer'>
	<div class='re-Footer-head'>
		<div class='re-Footer-content'>
			<h1 class='re-Footer-title'>Ronald Torres</h1>
			<h2 class='re-Footer-subTitle'>Front End Developer</h2>
		</div>
		<div class='re-Footer-enjoyer'>
			<span>React Enjoyer</span>
		</div>
	</div>
	<div class='re-Footer-legal'>
		<div class='re-Footer-disclaimer'>
			<p class='re-Footer-disclaimerTitle'>
				Â©{year} Ronald Torres. All rights reserved
			</p>
		</div>
	</div>
</footer>
