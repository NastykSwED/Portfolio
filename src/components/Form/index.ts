export * from './ContactFormField';
export * from './ContactFormInput';
export * from './ContactFormTextArea';
