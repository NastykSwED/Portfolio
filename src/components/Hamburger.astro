<div class='re-Hamburger'>
	<span class='re-Hamburger-bar'></span>
	<span class='re-Hamburger-bar'></span>
	<span class='re-Hamburger-bar'></span>
</div>

<script>
	const hamburgerElement = document.querySelector(
		'.re-Hamburger'
	) as HTMLElement;
	const headerNavbarElement = document.querySelector(
		'.re-Navbar'
	) as HTMLElement;

	const handleClick = () => {
		hamburgerElement.classList.toggle('isActive');
		headerNavbarElement.classList.toggle('isActive');
	};

	hamburgerElement.addEventListener('click', handleClick);
</script>
