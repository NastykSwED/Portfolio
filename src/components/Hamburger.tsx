import { useState } from 'react';

interface HamburgerProps {
	onClick: () => void;
}

export const Hamburger = (props: HamburgerProps) => {
	const { onClick } = props;

	const [isActive, setIsActive] = useState<boolean>(false);

	return (
		<div
			data-testid='Hamburger'
			onClick={() => {
				onClick();
				setIsActive(!isActive);
			}}
			className={`re-Hamburger ${isActive ? 'isActive' : ''}`}
		>
			<span className='re-Hamburger-bar'></span>
			<span className='re-Hamburger-bar'></span>
			<span className='re-Hamburger-bar'></span>
		</div>
	);
};
