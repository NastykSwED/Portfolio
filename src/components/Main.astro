---
import ContactSection from './Sections/Contact/ContactSection.astro';
import HomeSection from './Sections/Home/HomeSection.astro';
import ProjectsSection from './Sections/Projects/ProjectsSection.astro';
import SkillsSection from './Sections/Skills/SkillsSection.astro';
---

<main class='re-Main' id='re-Main'>
	<HomeSection />
	<SkillsSection />
	<ProjectsSection />
	<ContactSection />
</main>
