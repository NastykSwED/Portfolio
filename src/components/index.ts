export * from './Buttons';
export * from './Form';
export * from './Loaders';
export * from './Section';
export * from './Navbar/Navbar';
export * from './Hamburger';

