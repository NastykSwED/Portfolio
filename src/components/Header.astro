---
import Hamburger from './Hamburger.astro';
import Navbar from './Navbar/Navbar.astro';
---

<header class='re-Header' id='re-Header'>
	<div class='re-Header-wrapNavbar'>
		<h1 class='re-Header-logo'>Ronald Torres</h1>
		<Navbar />
		<Hamburger />
	</div>
</header>
