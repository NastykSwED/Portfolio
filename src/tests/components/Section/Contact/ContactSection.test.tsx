import { describe, it, expect } from 'vitest';

import { render } from '@testing-library/react';

import { ContactSection } from '../../../../components';

describe('Testing ContactSection component', () => {
	it('Should match the snapshot', () => {
		const { container } = render(<ContactSection />);

		expect(container).toMatchSnapshot();
	});
});
