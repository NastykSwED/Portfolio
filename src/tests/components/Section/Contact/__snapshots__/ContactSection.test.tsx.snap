// Vitest Snapshot v1

exports[`Testing ContactSection component > Should match the snapshot 1`] = `
<div>
  <section
    class="re-Main-ContactSection"
    id="re-ContactSection"
  >
    <div
      class="re-ContactSection-content"
    >
      <h1
        class="re-ContactSection-title"
      >
        Contact Me
      </h1>
      <h2
        class="re-ContactSection-subTitle"
      >
        Get in touch
      </h2>
    </div>
    <div
      class="re-ContactSection-bodyContent"
    >
      <div
        class="re-ContactSection-column"
      >
        <h3
          class="re-ContactSection-columnTitle"
        >
          Talk to me
        </h3>
        <div
          class="re-ContactSection-LayoutGrid"
        >
          <div
            class="re-ContactCard"
          >
            <span
              class="re-ContactCard-icon"
            >
              <svg
                class="Icon Icon--AiFillGithub u-CFI"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M511.6 76.3C264.3 76.2 64 276.4 64 523.5 64 718.9 189.3 885 363.8 946c23.5 5.9 19.9-10.8 19.9-22.2v-77.5c-135.7 15.9-141.2-73.9-150.3-88.9C215 726 171.5 718 184.5 703c30.9-15.9 62.4 4 98.9 57.9 26.4 39.1 77.9 32.5 104 26 5.7-23.5 17.9-44.5 34.7-60.8-140.6-25.2-199.2-111-199.2-213 0-49.5 16.3-95 48.3-131.7-20.4-60.5 1.9-112.3 4.9-120 58.1-5.2 118.5 41.6 123.2 45.3 33-8.9 70.7-13.6 112.9-13.6 42.4 0 80.2 4.9 113.5 13.9 11.3-8.6 67.3-48.8 121.3-43.9 2.9 7.7 24.7 58.3 5.5 118 32.4 36.8 48.9 82.7 48.9 132.3 0 102.2-59 188.1-200 212.9a127.5 127.5 0 0 1 38.1 91v112.5c.8 9 0 17.9 15 17.9 177.1-59.7 304.6-227 304.6-424.1 0-247.2-200.4-447.3-447.5-447.3z"
                />
              </svg>
            </span>
            <h4
              class="re-ContactCard-title"
            >
              Github
            </h4>
            <div
              class="re-ContactCard-wrapButton"
            >
              <a
                class="re-ContactCard-button"
                href="https://github.com/NastykSwED"
                rel="noopener noreferrer"
                target="_blank"
              >
                <button
                  class="Button Button--outline"
                  type="submit"
                >
                  Write Me
                  <span
                    class="Button-Icon"
                  >
                    <svg
                      class="Icon Icon--AiOutlineArrowRight"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 1024 1024"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M869 487.8L491.2 159.9c-2.9-2.5-6.6-3.9-10.5-3.9h-88.5c-7.4 0-10.8 9.2-5.2 14l350.2 304H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h585.1L386.9 854c-5.6 4.9-2.2 14 5.2 14h91.5c1.9 0 3.8-.7 5.2-2L869 536.2a32.07 32.07 0 0 0 0-48.4z"
                      />
                    </svg>
                  </span>
                </button>
              </a>
            </div>
          </div>
          <div
            class="re-ContactCard"
          >
            <span
              class="re-ContactCard-icon"
            >
              <svg
                class="Icon Icon--AiFillLinkedinu u-CFI"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zM349.3 793.7H230.6V411.9h118.7v381.8zm-59.3-434a68.8 68.8 0 1 1 68.8-68.8c-.1 38-30.9 68.8-68.8 68.8zm503.7 434H675.1V608c0-44.3-.8-101.2-61.7-101.2-61.7 0-71.2 48.2-71.2 98v188.9H423.7V411.9h113.8v52.2h1.6c15.8-30 54.5-61.7 112.3-61.7 120.2 0 142.3 79.1 142.3 181.9v209.4z"
                />
              </svg>
            </span>
            <h4
              class="re-ContactCard-title"
            >
              LinkedIn
            </h4>
            <div
              class="re-ContactCard-wrapButton"
            >
              <a
                class="re-ContactCard-button"
                href="https://www.linkedin.com/in/nastykswed/"
                rel="noopener noreferrer"
                target="_blank"
              >
                <button
                  class="Button Button--outline"
                  type="submit"
                >
                  Write Me
                  <span
                    class="Button-Icon"
                  >
                    <svg
                      class="Icon Icon--AiOutlineArrowRight"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 1024 1024"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M869 487.8L491.2 159.9c-2.9-2.5-6.6-3.9-10.5-3.9h-88.5c-7.4 0-10.8 9.2-5.2 14l350.2 304H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h585.1L386.9 854c-5.6 4.9-2.2 14 5.2 14h91.5c1.9 0 3.8-.7 5.2-2L869 536.2a32.07 32.07 0 0 0 0-48.4z"
                      />
                    </svg>
                  </span>
                </button>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="re-ContactSection-column"
      >
        <h3
          class="re-ContactSection-columnTitle"
        >
          Write to me
        </h3>
        <div
          class="re-ContactSection-wrapForm"
        >
          <form
            aria-label="form"
            class="re-ContactForm"
          >
            <div
              class="re-ContactForm-field"
            >
              <label
                class="re-ContactForm-label"
                for="userName"
              >
                Name
              </label>
              <input
                class="re-ContactForm-input"
                id="userName"
                name="userName"
                pattern="[A-Za-z ]{1,32}"
                placeholder="Enter your Name"
                required=""
                type="text"
                value=""
              />
              <span
                class="re-ContactForm-errorMessage"
              >
                The Name shouldn't include any special character or numbers
              </span>
            </div>
            <div
              class="re-ContactForm-field"
            >
              <label
                class="re-ContactForm-label"
                for="userEmail"
              >
                Email
              </label>
              <input
                class="re-ContactForm-input"
                id="userEmail"
                name="userEmail"
                pattern="[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+.[A-Za-z]{1,63}$"
                placeholder="Enter your Email"
                required=""
                type="text"
                value=""
              />
              <span
                class="re-ContactForm-errorMessage"
              >
                The Email should be valid!
              </span>
            </div>
            <div
              class="re-ContactForm-field"
            >
              <label
                class="re-ContactForm-label"
                for="userMessage"
              >
                Message
              </label>
              <textarea
                class="re-ContactForm-textArea"
                id="userMessage"
                name="userMessage"
                placeholder="Enter your Message"
                required=""
              />
              <span
                class="re-ContactForm-errorMessage"
              >
                The Message should be valid!
              </span>
            </div>
            <div
              class="re-ContactForm-wrapButton"
            >
              <button
                class="Button"
              >
                Send Message
                <span
                  class="Button-Icon"
                >
                  <svg
                    class="Icon Icon--AiOutlineSend"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    t="1569683742680"
                    version="1.1"
                    viewBox="0 0 1024 1024"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <defs />
                    <path
                      d="M931.4 498.9L94.9 79.5c-3.4-1.7-7.3-2.1-11-1.2-8.5 2.1-13.8 10.7-11.7 19.3l86.2 352.2c1.3 5.3 5.2 9.6 10.4 11.3l147.7 50.7-147.6 50.7c-5.2 1.8-9.1 6-10.3 11.3L72.2 926.5c-0.9 3.7-0.5 7.6 1.2 10.9 3.9 7.9 13.5 11.1 21.5 7.2l836.5-417c3.1-1.5 5.6-4.1 7.2-7.1 3.9-8 0.7-17.6-7.2-21.6zM170.8 826.3l50.3-205.6 295.2-101.3c2.3-0.8 4.2-2.6 5-5 1.4-4.2-0.8-8.7-5-10.2L221.1 403 171 198.2l628 314.9-628.2 313.2z"
                    />
                  </svg>
                </span>
              </button>
            </div>
            <div
              style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
            />
          </form>
        </div>
      </div>
    </div>
  </section>
</div>
`;
